build.gradle

```diff
@@ -17,6 +17,7 @@ buildscript {
 
 plugins {
     id 'com.gradle.build-scan' version '1.0'
+    id 'org.jetbrains.kotlin.jvm' version '1.1.1'
```

Comment:
I really, really like the idea of adding tests in Kotlin! This makes Mockito coverage much better.

I do have one request. Can we move the Kotlin test to a separate subproject? I would feel safer if we avoid using Kotlin plugin for the Gradle subproject that builds main distribution of Mockito. I haven't scrutinized the Kotlin plugin but plugins can add dependencies and do other interesting things that can alter the binary or pom of the distribution.

Also, it's safer for the core tests if they are decoupled from Kotlin and run without Kotlin libraries on the classpath.